### ISSUE TITLE
Parse error on raw inline literals containing `#` in `let` expression

### ISSUE THREAD

**Issue by ntjess** [jgm/typst-hs#22]
```typst
#let x = `#`
// Also happens with ```#```
```

Parsing produces
```console
unexpected "`"
expecting "//", "/*", "none", "auto", "true", "false", "0b", "0x", "0o", digit, ".", "\"", "let", "set", "show", "if", "while", "for", "import", "include", "break", "continue", "return", "(", "$", "<", "{" or "["    
```

**Comment by jgm** [comment]
Well it errors also with
```
#let x = `a`
```
I didn't know you could use the syntax this way; is it supposed to be the same as
```
#let x = [`a`]
```
?
Pointers to documentation on this?

**Comment by ntjess** [comment]
Nothing in the documentaiton, and a [github search](https://github.com/search?q=path%3A*.typ++%2F%5C%23let.*%60%2F&type=code&query=repo%3Atypst%2Ftypst+path%3A*.typ+%2F%23let.*%3D%60%2F) indicates that the official typst repo doesn't do this in their tests/codebases either

**Comment by jgm** [comment]
It would be good to get them to either commit to this and document it, or remove it, I think.

**Comment by ntjess** [comment]
I found https://typst.app/docs/reference/foundations/regex/#constructor-regex which recommends using backticks in code mode to avoid escaping regex backslashes

**Comment by jgm** [comment]
Fixed by 3512d626731c8b924f0705083e7a5508566f9a47
